src/getProxyAgentFromUri.ts

```diff
@@ -0,0 +1,93 @@
+import { Agent } from "http"
```

Comment:
This is partly lifted from [`request#getProxyFromUri`](https://github.com/request/request/blob/b12a6245d9acdb1e13c6486d427801e123fdafae/lib/getProxyFromURI.js)  but: in TypeScript, with support for reading from the Yarn config, and returning an `Agent` instead of a proxy string (for inter-operation with `node-fetch`).

Reply:
Sure thing, I've never used this stuff so I'll believe it 👍